src/platform/lumin-runtime/mxs-base-app.js

```diff
@@ -4,16 +4,37 @@ import { AppPrismController } from './controllers/app-prism-controller.js';
 import { ReactMagicScript } from '../../react-magic-script/react-magic-script.js';
 
 import executor from './utilities/executor.js';
-import { logInfo } from '../../util/logger.js';
+import { logInfo, logWarning } from '../../util/logger.js';
 
 export class MxsBaseApp {
   constructor(appComponent) {
       this._app = appComponent;
       this._prisms = [];
       this._prismControllers = [];
+      this._eventHandlers = {
+        // The rendering process hasn't started yet
+        // OnAppStartData will be provided on demand as property.
+        // onAppStart: [],
+        onAppPause: [],
+        onAppResume: []
+      };
+
+      this._onAppStartData;
   }
 
-  onAppStart(arg) {
+  get OnAppStartData () {
+    return this._onAppStartData;
+  }
+
+  onAppStart(args) {
+    this._onAppStartData = { 
+      uri: args.getUri(),
+      isInternetConnected: typeof this.isInternetConnected === 'function' ? this.isInternetConnected() : undefined,
```

Comment:
Everything besides uri seems unrelated to app start data. Also, besides isShareableApp (which I believe reflects a manifest field), the rest could change at any moment so it seems problematic to take a snapshot of them only when onAppStart is called.

Reply:
I agree! The connectivity might change. There is an event `WifiStatusEvent` which the developer can use to get updates on the connectivity state. The goal was to provide some initial app status which there is no other way to provide until an event occurs.

Reply:
This seems unintuitive but I don't know the best way to handle this in React so I'll resolve the change request.

Reply:
In iOS, there is `Reachability` framework that notifies about WiFi/Bluetooth/Cell network changes. We could provide a similar module as we did with Platform. But this is another task, isn't it?